opennav_docking/src/navigator.cpp

```diff
@@ -66,19 +69,37 @@ void Navigator::goToPose(
       future_goal_handle, 2s) == rclcpp::FutureReturnCode::SUCCESS)
   {
     auto future_result = nav_to_pose_client_->async_get_result(future_goal_handle.get());
-    if (executor_.spin_until_future_complete(
-        future_result, timeout) == rclcpp::FutureReturnCode::SUCCESS)
-    {
-      auto result = future_result.get();
-      if (result.code == rclcpp_action::ResultCode::SUCCEEDED && result.result->error_code == 0) {
-        return;  // Success!
+
+    while (rclcpp::ok()) {
+      if (isPreempted()) {
+        auto cancel_future = nav_to_pose_client_->async_cancel_goal(future_goal_handle.get());
+        executor_.spin_until_future_complete(cancel_future, 1s);
+        throw opennav_docking_core::FailedToStage("Navigation request to staging pose preempted.");
+      }
+
+      if (node->now() - start_time > max_staging_duration) {
+        nav_to_pose_client_->async_cancel_goal(future_goal_handle.get());
+        throw opennav_docking_core::FailedToStage("Navigation request to staging pose timed out.");
```

Comment:
Similarly, this should be removed, or use the recurse doesn't do anything. Only the `isPreempted` should throw

Reply:
Can we retry after exceeding `max_staging_duration`?

Reply:
I mean, does it make sense? Depends on what this duration refers to

Reply:
Good point, I think that duration should refer to the full sequence of navigating to pose. If it fails, we should retry within that duration, but multiple retries should still fit into that single duration (from the application's perspective, retrying navigation is a 'detail' to accomplish the step)

Reply:
I think to make that work, we'd have to (a) mess with the retry to not be recursive so we can track the total duration among the retry or (b) have the method given the 'end time'/duration remaining to use, rather than the full duration so when called recursively it doesn't try to use the full duration both times.